### DISCUSSION TITLE
Off shelf shifters

### DISCUSSION THREAD

**Discussion by chinasaur** [doudar/SmartSpin2k#213] [Q&A] [answered]
Anyone try something like [this for shifting](https://www.amazon.com/dp/B08T1WBQHQ/)? Aluminum, designed for motorcycle handlebar, 2 momentary switches. I'd like to try, but may need a bit larger than 25 mm for the c6 / ic4 / ic8 handlebars.

(Double check it's the momentary version and note the review describing weird wire coloring.)

**Comment by doudar** [comment, marked as answer]
Wow! Nice find.

it looks like they would work perfectly.

**Reply by chinasaur** [reply]
So, actually for the C6 / IC4 bike, I realized there are up/down buttons right there on the built-in console. These hardly do anything useful in normal operation, so it would be great to use them for shifting. Anyone have any info on tapping into this on the console PCB?

**Reply by doudar** [reply]
> So, actually for the C6 / IC4 bike, I realized there are up/down buttons right there on the built-in console. These hardly do anything useful in normal operation, so it would be great to use them for shifting. Anyone have any info on tapping into this on the console PCB?

If you sent some pictures of the PCB I might be able to point you in the right direction.

**Reply by chinasaur** [reply]
Thanks! Here's a photo of the front side of the board. The buttons are on the other side, so I probably need to take it apart further? The ribbon at the bottom I assume goes to the buttons / LCD, and I think the power connection at bottom right goes to the console. The nuvoTon next to the speaker is the ARM processor. I have close-ups and have identified most of the chips on the board if any other info would be helpful.

![image](https://user-images.githubusercontent.com/68619/119434143-6b087a00-bccc-11eb-8c09-934209a43107.png)

**Reply by chinasaur** [reply]
I guess this R10a through R14a are possible, since they go to pins on the ARM and they go into vias on the other side. And there are 5 of them and 5 buttons. Do button lines usually have resistors in series? These are probably big enough for me to get a probe onto.

**Reply by chinasaur** [reply]
Success; pushing the button GNDs the line.

- R10a: Up button
- R11a: Down button
- R12a: Circle arrow (back?) button
- R13a: Menu button
- R14a: Bluetooth button

Not sure how to wire into this though. I guess I can just solder a line onto the resistor or the via?

**Reply by doudar** [reply]
Check voltage on both (esp32 and the IC4) first to make sure they are exactly the same 3.3v when not pressed and then yes, you *should* be able to just connect them. For a little extra safety I'd also add a small resistor (~100ohm) between the IC4 and ESP32 to help with minor voltage variations (and the fact the line would now have two pull up resistors) and then find a common ground somewhere to connect also.

If the 100ohm resistor ends up being too large (ESP32 doesn't recognize button presses), it would be best to incrementally lower the resistor value until it registers the presses. 

If you wanted the 100% safe solution, you could order a little optoisolator and use it between the two. 

https://www.sparkfun.com/products/9118

"HV" on that board would just get connected to the 3.3v on the ESP32. 

Good luck! Don't burn up the IC4!

**Reply by chinasaur** [reply]
Thanks! Yes, too bad the board seems not easily replaceable without a support call. OTOH, there's nothing this console does that an ESP32 can't (except aggressive beeping), especially with SmartSpin2k available. (The resistance and cadence sensors are easy to read from the header in the lower left.) Still, I will be in trouble if I break it; hence my reluctance to disassemble further. I was hoping it would be a little easier to tap into the buttons like it is for the sensors. I might poke around the unpopulated pads on the top of the board, but not likely to be useful.

In terms of connecting, the IC4 gets 9V 1.5A nominal from the adapter, interesting since it's enough for a small motor and the board doesn't seem to need much of the power. The board runs 5V (regulator lower left). GND is available from the harness header lower left, or either of the red open headers. 5V is available from the harness or the 5-pin red open header. If I do go further in this direction, I guess just an 8-channel level shifter or something makes sense. It would be nice to get at the harness from the external circular connector, but the connector seems hard to track down.

At this point I'll get the rest of the project together and then come back to these refinements.

**Reply by chinasaur** [reply]
Brief update, I'm pretty sure the ribbon cable at bottom JK1e is also the buttons. R1-5e are 10k pull-ups to the 5V regulator. The 6th conductor in the cable I guess is GNDed. I'll check signals to confirm when I have a chance.

So hopefully this provides a simple, clean way to access things. The connector is some kind of FCC/FPC 1 mm pitch slide lock bottom conductor, but not sure the specific part.
